src/main/scala/com/gu/octopusthrift/models/OctopusPayload.scala

```diff
@@ -2,23 +2,37 @@ package com.gu.octopusthrift.models
 
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
-
-/** This model represents part of the payload received from Octopus */
-case class OctopusPayload(
-  `type`: String,
+case class OctopusBundleCacheData(
   bundles: Option[Array[OctopusBundle]],
-  data: Option[OctopusBundle],
   thismessageindex: Option[Int],
   totalmessages: Option[Int])
+object OctopusBundleCacheData {
+  implicit val bundleCacheDataFormat = Json.format[OctopusBundleCacheData]
+}
+
+/**
+ * The following models represent the possible payload shapes we can receive from Octopus
+ *  - the OctopusBundleCachePayload represents a daily snapshot of Octopus data, containing multiple story bundles
+ *  - the OctopusSingleBundlePayload represents an update to a single story bundle
+ */
+case class OctopusBundleCachePayload(`type`: String, data: Option[OctopusBundleCacheData])
+object OctopusBundleCachePayload {
+
+  val typeReads = Reads.StringReads.filter(str => {
+    str.matches("bundlecache")
+  })
 
-object OctopusPayload {
-  implicit val reads: Reads[OctopusPayload] =
-    ((__ \ "type").read[String] and (__ \ "bundles").readNullable[Array[OctopusBundle]] and (__ \ "data")
-      .readNullable[OctopusBundle] and (__ \ "thismessageindex").readNullable[Int] and (__ \ "totalmessages")
-      .readNullable[Int])(OctopusPayload.apply _)
+  implicit val reads: Reads[OctopusBundleCachePayload] =
+    ((__ \ "type").read[String](typeReads) and (__ \ "data")
+      .readNullable[OctopusBundleCacheData])(OctopusBundleCachePayload.apply _)
+
+  implicit val writes: Writes[OctopusBundleCachePayload] =
+    ((__ \ "type").write[String] and (__ \ "data").writeNullable[OctopusBundleCacheData])(
+      unlift(OctopusBundleCachePayload.unapply))
+
+}
 
-  implicit val writes: Writes[OctopusPayload] =
-    ((__ \ "type").write[String] and (__ \ "bundles").writeNullable[Array[OctopusBundle]] and (__ \ "data")
-      .writeNullable[OctopusBundle] and (__ \ "thismessageindex")
-      .writeNullable[Int] and (__ \ "totalmessages").writeNullable[Int])(unlift(OctopusPayload.unapply))
+case class OctopusSingleBundlePayload(`type`: String, data: Option[OctopusBundle])
+object OctopusSingleBundlePayload {
+  implicit val singleBundleFormat = Json.format[OctopusSingleBundlePayload]
```

Comment:
Is this a nicer way of creating a reads for a case class? Should we be doing this in more places?

Reply:
Yes, it's a [`Format`](https://www.playframework.com/documentation/2.8.x/ScalaJsonCombinators#Format). We may be able to use it in other places, but I don't _think_ it's usable quite as easily as this in situations when the case class parameters don't match the JSON keys, e.g. in the OctopusArticle when we read `for_publication` into `forPublication`. We may be able to combine our existing reads and writes into a format, though.

Reply:
As I've suggested previously, have a set of input case classes that directly match your expected input and then use them to populate your output case classes. Manual boiler plate is ideally avoided.

I'd also express in code whether you expect to read or wrote from JSON. In this case I imagine you are only reading so use `Json.reads` rather than `Json.format`.

Reply:
Good point, @sihil - I've updated the models.